app/src/main/java/ch/epfl/sdp/appart/scrolling/card/ApartmentCardAdapter.java

```diff
@@ -0,0 +1,104 @@
+package ch.epfl.sdp.appart.scrolling.card;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import ch.epfl.sdp.appart.R;
+import ch.epfl.sdp.appart.scrolling.AnnounceActivity;
+import ch.epfl.sdp.appart.scrolling.ScrollingActivity;
+
+/**
+ * Based on the following tutorial : https://developer.android.com/codelabs/basic-android-kotlin-training-recyclerview-scrollable-list
+ */
+public class ApartmentCardAdapter extends RecyclerView.Adapter<ApartmentCardAdapter.CardViewHolder> {
+
+    private final List<ApartmentCard> cards;
+    private final Context context;
+
+    public ApartmentCardAdapter(Activity context, List<ApartmentCard> cards) {
```

Comment:
Yes I had a design problem because I start the Announce activity from the Adapter instead of the Scrolling Activity. This is because the cardview is managed in the viewholder class and therefore the click listener is inside the viewholder class.

Reply:
I would just do this since you assign context to this.context which is already of type Context. Or maybe I didn't understand what you said.

```suggestion
    public ApartmentCardAdapter(Context context, List<ApartmentCard> cards) {
```

Reply:
Here are the problematic lines :
```
holder.cardImageView.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Intent intent = new Intent(context, AnnounceActivity.class);
                context.startActivity(intent);
            }
        });
```